crm/crm/doctype/customer_feedback/customer_feedback.py

```diff
@@ -204,3 +213,20 @@ def postprocess(source, target):
 			}
 		},
 	}, postprocess=postprocess)
+
+def validate(doc, method=None):
+    if doc.feedback_sub_type:
+        valid = frappe.db.exists("Feedback Sub Type", {
+			"name": doc.feedback_sub_type,
+			"feedback_type": doc.feedback_type
+		})
+        if not valid:
+            frappe.throw("Selected Feedback Sub Type is not valid for the selected Feedback Type.")
+
+    if doc.feedback_status:
+        valid_status = frappe.db.exists("Feedback Status", {
+			"name": doc.feedback_status,
+			"feedback_type": doc.feedback_type
+		})
+        if not valid_status:
+            frappe.throw("Selected Feedback Status is not valid for the selected Feedback Type.")
```

Comment:
- no need to hook onto your own app's doctype, move to controller
- fix indentation
- translate strings